cpp/CMakeLists.txt

```diff
@@ -35,6 +35,8 @@ endif()
 ###################################################################################################
 # - compiler options ------------------------------------------------------------------------------
 
+option(CMAKE_CXX11_ABI "Enable the GLIBCXX11 ABI" OFF)
```

Comment:
I think this option should only be available and set inside `if(CMAKE_COMPILER_IS_GNUCXX)`. It is a pointless flag on non-linux-gnu platforms.

@mt-jones agree?

Reply:
Specifically, I suggest structuring it like this (note I also reversed the if condition to put the positive condition first).

```cmake
if(CMAKE_COMPILER_IS_GNUCXX)
    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")

    option(CMAKE_CXX11_ABI "Enable the GLIBCXX11 ABI" OFF)
    if(CMAKE_CXX11_ABI)
        message(STATUS "CUDF: Enabling the GLIBCXX11 ABI")
    else()
        message(STATUS "CUDF: Disabling the GLIBCXX11 ABI")
        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=0")
        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=0")
    endif(CMAKE_CXX11_ABI)
endif(CMAKE_COMPILER_IS_GNUCXX)```